musicbot/entry.py

```diff
@@ -319,6 +324,29 @@ def is_exe(fpath):
                 # Move the temporary file to it's final location.
                 os.rename(unhashed_fname, self.filename)
 
+        # Get duration from the file after downloaded
+        args = [
+            'ffprobe', 
+            '-i', self.filename, 
+            '-show_entries', 'format=duration', 
+            '-v', 'quiet', 
+            '-of', 'csv="p=0"'
+        ]
+
+        output = await self.run_command(args)
+        output = output.decode("utf-8")
+
+        try:
+            self.duration = float(output)
+        except ValueError:
+            # @TheerapakG: If somehow it is not string of float
+            if not self.duration:
+                log.error('Cannot extract duration of downloaded entry, invalid output from ffprobe.'
+                          'This does not affect the ability of the bot. However, estimated time for this entry'
+                          'will not be unavailable and estimated time of the queue will also not be available'
+                          'until this entry got removed.'
+                          'entry file: {}'.format(self.filename))
+
```

Comment:
This is an expensive thing to run every time for every song, maybe we should only do this if the duration is None?

Reply:
Is there no other way to fetch the duration of the file than to do this? I’d be fine with something that didn’t rely on running an ffmpeg command on the cli and will be fine with approving this as it’ll rarely activate but maybe looking into a less hacky solution now will do us better for preventing future technological debt.

Reply:
Does this suffice an additional pip requirement?

Reply:
I'd prefer not to tbh if it can be avoided

Reply:
I'm afraid that that is impossible with something we currently have in requirements. This would need something like pymediainfo.